diffrax/_integrate.py

```diff
@@ -124,6 +125,8 @@ def _term_compatible(
     contr_kwargs: PyTree[dict],
 ) -> bool:
     error_msg = "term_structure"
+    if isinstance(y, KLState):
+        y = y.y
```

Comment:
I don't think this can be allowed unfortunately :(
The contents of `_integrate.py` try to avoid ever special-casing particular inputs. Rather, they try to simply interact with the abstractions provided by ABCs.
(As you scroll through this file you'll find that there are a few exceptions to this, but I think they are all exclusively minor performance optimisations.)

The general principle is that someone should be able to implement equivalent functionality on top of Diffrax, without ever needing to change its internals.

Reply:
The reason this was here was because the term checking would fail since the term itself is unaware of the KL state. I was thinking of ways around this, and it seems like the easiest way would be to require the term to be aware of it. 

Much of the design was focused on the user not actively controlling the KLstate, but maybe it makes more sense to require them to provide it and for their term to actually consume it, rather than trying to totally hide it and make it so they don't have to do anything. 

I will push a change enforcing that requirement.